### PR TITLE
Issue/21 language system changes in 3.2

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Depends on #36 

@MateBartus please review https://github.com/phpbb/documentation/commit/6bf972a5a69aefde375ad737a10df2491540cd0e

### CODE REVIEW

File: development/language/usage.rst [hunk 1]
```diff
@@ -8,48 +8,96 @@ also a requirement for Extensions to use said system for all output presented to
...
+
+Since ``phpbb\user::setup()`` must only be called once,
+``phpbb\language\language::add_lang()`` has to be used, to load additional
+language files, after ``phpbb\user::setup()`` has already been called.
+
+Loading from an extension
+-------------------------
+
+To load a file from an extension
+you need to use ``phpbb\language\language::add_lang_ext()`` which takes
```
Comment (MateBartus) [inline development/language/usage.rst]: ``phpbb\language\language::add_lang()`` first argument array of the language files, second argument is the name of the extension

File: development/language/usage.rst [hunk 2]
```diff
@@ -8,48 +8,96 @@ also a requirement for Extensions to use said system for all output presented to
...
+the vendor + extension name as first argument and the array of language files as
+a second argument.
 
 .. code-block:: php
 
-    $message .= '<br /><br />' . $user->lang('MY_TRICKY_LANGUAGE_KEY', '<a href="' . append_sid("{$phpbb_root_path}mypage.$phpEx") . '">', '</a>');
-    trigger_error($message);
+    $language->add_lang_ext('acme/demo', array('demo'));
```
Comment (MateBartus) [inline development/language/usage.rst]: ``$language->add_lang(array('demo'), 'acme/demo');``

File: development/language/usage.rst [hunk 3]
```diff
@@ -57,17 +105,22 @@ Using the Language System in template files
 Language entries in
 `Using the Template System <https://wiki.phpbb.com/Using_the_phpBB3.0_Template_System>`_
 are a major improvement since phpBB 3.0. There is no longer a need to manually
-assign these in the php file; loaded language files will be automatically used.
+assign these in the PHP file; language entries of loaded language files can be
+used automatically.
+
 To use the language entry with the key ``MY_KEY`` in a template file, just write
-``{L_MY_KEY}`` in the template - it's as simple as that.
```
Comment (MateBartus) [inline development/language/usage.rst]: I think you can use ``{L_MY_KEY}`` as well. @Nicofuma?

File: development/language/usage.rst (hunk 2 above)
Comment (MateBartus) [inline development/language/usage.rst]: btw  ``$language->add_lang('demo', 'acme/demo');`` is also valid

File: development/language/usage.rst (hunk 3 above)
Comment (Nicofuma) [inline development/language/usage.rst]: `{L_MY_KEY}` or `{{ lang('MY_KEY') }}` in full twig syntax

Comment (nickvergessen) [inline development/language/usage.rst]: `{{ lang('MY_KEY') }}` works also in 3.1?

Comment (Nicofuma) [inline development/language/usage.rst]: AFAIK yes

**Comment by nickvergessen** [comment]
@CHItA updated

File: development/language/usage.rst [hunk 4]
```diff
@@ -84,22 +85,22 @@ pass the name with the subdirectory but without extension as argument of setup.
...
-after ``phpbb\user::setup()`` has already been called.
+``phpbb\language\language::add_lang()`` has to be used, to load additional
+language files, after ``phpbb\user::setup()`` has already been called.
 
 Loading from an extension
 -------------------------
 
 To load a file from an extension
-you need to use ``phpbb\user::add_lang_ext()`` which takes
+you need to use ``phpbb\language\language::add_lang_ext()`` which takes
```
Comment (CHItA) [inline development/language/usage.rst]: add_lang() as well for exts, as in the examples below

Comment (nickvergessen) [inline development/language/usage.rst]: hmpf must have slipped through

**Comment by VSEphpbb** [comment]
Add a bit about the new lang system replacement for LA_KEY by doing `{{ lang('KEY')|e('js') }}`
Other possible options:
`{{ lang('KEY')|escape('js') }}`
`{{ lang('KEY')|e('addslashes') }}`
`{{ lang('KEY')|escape('addslashes') }}`

addslashes is our own custom func that simply addslashes the string. 
js is a native twig function that escapes a string for the JavaScript context.

**Comment by nickvergessen** [comment]
@VSEphpbb since those comments need to go into 3.1 aswell, please create a new issue (and feel free to prepare a patch ;) )

File: development/language/usage.rst (hunk 4 above)
Comment (nickvergessen) [inline development/language/usage.rst]: Updated

**Pull request merged**